### PR TITLE
restore_reflections() use renamed motor axes if so defined

### LINKED ISSUE #215: restore_reflections(orientation, fourc) cannot find renamed positioner
As [reported](https://github.com/APS-4ID-POLAR/ipython-polar/issues/146#issuecomment-1143809641) by @gfabbris, `restore_reflections(orientation, fourc)` could not find `theta` in the reflection dict:

```py
  {'reflection': {'h': 1.0, 'k': -3.0, 'l': -1.0},
   'flag': 1,
   'wavelength': 1.1169734383241103,
   'position': {'omega': -8.208399999999983,
    'chi': -47.47651999999994,
    'phi': -5.684341886080802e-14,
    'tth': 25.09473789610388},
   'orientation_reflection': False},
```

when the positioners were [renamed](https://github.com/APS-4ID-POLAR/ipython-polar/blob/fa20b5420b2b2fa51c577c861bd589abcfac7967/profile_bluesky/startup/instrument/devices/huber.py#L84-L87) such as 

```py
fourc.calc.physical_axis_names = {'omega': 'theta',
                                  'chi': 'chi',
                                  'phi': 'phi',
                                  'tth': 'tth'}
```

### PR DESCRIPTION
**Pull request by prjemian**
- fix #215

### CODE REVIEW

File: hkl/tests/test_util.py [hunk 1]
```diff
@@ -6,6 +6,75 @@
 NO_SUCH_PACKAGE_NAME = "no-such-package"
 
 
+@pytest.fixture(scope="function")
+def cat():
+    import databroker
+
+    yield databroker.temp().v2
```
Comment (mrakitin) [inline hkl/tests/test_util.py:13]: I typically do `yield` to perform some cleanup after that, but I think it's a good pattern in general.
https://docs.pytest.org/en/6.2.x/fixture.html has different approaches depending on the needs.

File: hkl/tests/test_util.py [hunk 2]
```diff
@@ -6,6 +6,75 @@
...
+
+
+def crystal_setup(diffractometer):
+    from hkl import Lattice
+
+    diffractometer.calc.wavelength = 1.0
+    a0 = 5.4321
+    # fmt: off
+    diffractometer.calc.new_sample(
+        "vibranium",
```
Comment (mrakitin) [inline hkl/tests/test_util.py:62]: +1 for the material name! Good for tests.

**Review by mrakitin — APPROVED** [review]
This one looks good to me.

File: hkl/tests/test_util.py (hunk 2 above)
Comment (prjemian) [reply hkl/tests/test_util.py:62]: You noticed!  Who can dispute the crystal lattice, anyway?

**Pull request merged**